File: frontend/src/types/db/post.ts
import { z } from "zod";

const PostSchema = z.object({
    id: z.number(),
    user_account_id: z.number(),
    image_id: z.number(),
    title: z.string(),
    content: z.string(),
    created_at: z.string().datetime(),
    updated_at: z.string().datetime(),
})

type Post = z.infer<typeof PostSchema>;

export { PostSchema };
export type { Post };

File: backend/internal/domain/entity/post.go
package entity

import (
	"time"
)

type Post struct {
	ID            int64
	UserAccountID int64
	ImageID       int64
	Title         string
	Content       string
	Likes         int
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

func NewPost(userAccountID, imageID int64, title, content string) *Post {
	now := time.Now()
	return &Post{
		UserAccountID: userAccountID,
		ImageID:       imageID,
		Title:         title,
		Content:       content,
		Likes:         0,
		CreatedAt:     now,
		UpdatedAt:     now,
	}
}

func (p *Post) IncrementLikes() {
	p.Likes++
	p.UpdatedAt = time.Now()
}